include/envoy/common/platform.h

```diff
@@ -248,19 +248,15 @@ struct mmsghdr {
 #endif // __ANDROID_API__ < 24
 #endif // ifdef __ANDROID_API__
 
-#ifdef __linux__
-#define SUPPORTS_PTHREAD_GETNAME_NP 1
-#endif
-
 // https://android.googlesource.com/platform/bionic/+/master/docs/status.md
 // ``pthread_getname_np`` is introduced in API 26
-#ifdef __ANDROID_API__
-#if __ANDROID_API__ > 26
-#define SUPPORTS_PTHREAD_GETNAME_NP 1
-#endif // __ANDROID_API__ > 26
-#endif // ifdef __ANDROID_API__
-
-// Ensure `SUPPORTS_PTHREAD_GETNAME_NP` is set
-#ifndef SUPPORTS_PTHREAD_GETNAME_NP
 #define SUPPORTS_PTHREAD_GETNAME_NP 0
-#endif
+#if defined(__ANDROID_API__)
+#if __ANDROID_API__ >= 26
+#undef SUPPORTS_PTHREAD_GETNAME_NP
+#define SUPPORTS_PTHREAD_GETNAME_NP 1
+#endif // __ANDROID_API__ >= 26
+#elif defined(__linux__)
+#undef SUPPORTS_PTHREAD_GETNAME_NP
+#define SUPPORTS_PTHREAD_GETNAME_NP 1
+#endif // defined(__ANDROID_API__)
```

Comment:
```suggestion
#define SUPPORTS_PTHREAD_NAMING 0
#if defined(__linux__) && !defined(__ANDROID_API__) || __ANDROID_API__ >= 26
#undef SUPPORTS_PTHREAD_NAMING
#define SUPPORTS_PTHREAD_NAMING 1
#endif
```
How about this?

The renaming suggestion to `SUPPORTS_PTHREAD_NAMING` is because this macro name implies we only care about the existence of when method, when in fact we're using/guarding two below.

Reply:
We're currently only guarding against the one method right now

Reply:
> We're currently only guarding against the one method right now I believe

Right, `pthread_getname_np` is the only one we're missing, so I think the existing name more closely reflects _why_ we're guarding with this conditional.

Also, your suggestion is not equivalent to what I have on this branch @goaway. I'm not requiring `__linux__`, but instead am prioritizing the value of Android's API version if it's specified (which is more similar to what was in place prior to this change).

Reply:
https://github.com/envoyproxy/envoy/blob/8a21fc79914db909d98b326b2c622aed3d94b6f8/source/common/common/posix/thread_impl.cc#L62

Reply:
I'm aware that `pthread_setname_np` is being called within the compiler conditional, but `pthread_getname_np` is the function that is unavailable, as detailed in the docs linked inline. AFAICT `pthread_setname_np` is gated by the conditional because it shouldn't be called if `pthread_getname_np` is unavailable.

Reply:
We use the guard to generally disable the logic around pthread naming - both set and get.

The boolean logic I posted above is equivalent to what you have, it just reduces the number of conditionals.

Reply:
I don't have that strong of an opinion here. Renamed to what you requested.